src/analysis/processing/qgsalgorithmrandompointsonlines.cpp

```diff
@@ -0,0 +1,264 @@
+/***************************************************************************
+                         qgsalgorithmrandompointsonlines.cpp
+                         ---------------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Håvard Tveite
+    email                : havard dot tveite at nmbu dot no
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "qgsalgorithmrandompointsonlines.h"
+#include "random"
+
+///@cond PRIVATE
+
+QString QgsRandomPointsOnLinesAlgorithm::name() const
+{
+  return QStringLiteral( "randompointsonlines" );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::displayName() const
+{
+  return QObject::tr( "Random points on lines" );
+}
+
+QStringList QgsRandomPointsOnLinesAlgorithm::tags() const
+{
+  return QObject::tr( "random,points,lines,seed,attributes,create" ).split( ',' );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::group() const
+{
+  return QObject::tr( "Vector creation" );
+}
+
+QString QgsRandomPointsOnLinesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorcreation" );
+}
+
+void QgsRandomPointsOnLinesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input line layer" ), QList< int >() << QgsProcessing::TypeVectorLine ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "POINTS_NUMBER" ), QObject::tr( "Number of points for each feature" ), QgsProcessingParameterNumber::Integer, 1, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "MIN_DISTANCE" ), QObject::tr( "Minimum distance between points" ), 0, QStringLiteral( "TARGET_CRS" ), true, 0 ) );
+  addParameter( new QgsProcessingParameterCrs( QStringLiteral( "TARGET_CRS" ), QObject::tr( "Target CRS" ), QStringLiteral( "ProjectCrs" ), false ) );
```

Comment:
Which other algorithm? It should only be exposed in very limited circumstances. By design we aim to make native algorithms as modular and "stripped" back as possible, so we don't have options like this and instead require users to run the dedicated reprojection algorithm on the result.

Reply:
The *Random points in extent* algorithm.

Reply:
Ah - that's there for compatibility. In earlier QGIS versions extent parameters were purely numeric, and didn't have any knowledge of their associated CRS. It could potentially be hidden from that algorithm now that extent values include their CRS.